### PR TITLE
[ios] Fixed non-working API calls

### LINKED ISSUE #510: [ios] geo:// and om:// url schemes do not work on a cold start
If the app is already launched, then points are displayed correctly.

### LINKED ISSUE #5107: iOS: Open URI only works the first time
**Describe the issue**
When I open the app via a url like`https://omaps.app/map?v=1&ll=50.08680725097656,14.4188871383667` and then open it again from another url (like `https://omaps.app/map?v=1&ll=56.62826167346705,4.031875610350426`) then the second time it zooms to the location of the first link again. Until I close and reopen the app and only click the second link, then it works.

**Steps to reproduce**
1. open https://omaps.app/map?v=1&ll=50.08680725097656,14.4188871383667
2. leave organic maps, but don't close it from the app overview
3. open https://omaps.app/map?v=1&ll=56.62826167346705,4.031875610350426
4. See that it zooms to the point of the first url again

**Expected behavior**
It zooms to the second point when I click the second link. without force closing the app in between.

**System information:**
- Operating system and its version: iOS 16
- Organic Maps version: 2023.04.02-6
- Device Model: iPhone SE 2020

### PR DESCRIPTION
**Pull request by biodranik**
Fixes #5107
Fixes #510

Need to test it with https://omaps.app/api and other examples.

### CODE REVIEW

File: drape_frontend/visual_params.cpp [hunk 1]
```diff
@@ -217,6 +217,7 @@ int GetTileScaleBase(ScreenBase const & s)
 int GetTileScaleBase(m2::RectD const & r)
 {
   double const sz = std::max(r.SizeX(), r.SizeY());
+  ASSERT_NOT_EQUAL(sz, 0., ("Rect should not be a point:", r));
```
Comment (vng) [inline drape_frontend/visual_params.cpp:220]: nit: IMO the comment in assert is a tautology.
ASSERT_GREATER(sz, 0, ())
is enough

File: map/mwm_url.cpp [hunk 2]
```diff
@@ -376,9 +376,12 @@ void ParsedMapApi::Reset()
...
 {
   auto const & markIds = m_bmManager->GetUserMarkIds(UserMark::Type::API);
-  if (markIds.size() == 1 && m_zoomLevel >= 1)
+  if (markIds.size() == 1)
   {
-    scale = min(static_cast<double>(scales::GetUpperComfortScale()), m_zoomLevel);
+    if (m_zoomLevel >= 1)  // 0 means uninitialized/not passed to the API.
+      scale = min(static_cast<double>(scales::GetUpperComfortScale()), m_zoomLevel);
+    else
+      scale = static_cast<double>(scales::GetUpperComfortScale());
```
Comment (vng) [inline map/mwm_url.cpp:384]: nit: prettify
```
scale = static_cast<double>(scales::GetUpperComfortScale());
if (m_zoomLevel >= 1 && scale < m_zoomLevel)
  scale = m_zoomLevel;
```

**Review by vng — APPROVED** [review]

File: map/mwm_url.cpp (hunk 2 above)
Comment (biodranik) [reply map/mwm_url.cpp:384]: IMO your version is less readable and has one additional write in popular case.

File: drape_frontend/visual_params.cpp (hunk 1 above)
Comment (biodranik) [reply drape_frontend/visual_params.cpp:220]: Not every reader will understand that it's zero in a point case :)

**Pull request merged**